src/components/AppLayout/FilterBar/Filter.tsx

```diff
@@ -0,0 +1,129 @@
+import {
+  FilterContent,
+  FilterElement,
+  FilterErrorMessages,
+  IFilter,
+  InvalidFilters,
+} from "@dashboard/components/Filter";
+import useFilter from "@dashboard/components/Filter/useFilter";
+import { extractInvalidFilters } from "@dashboard/components/Filter/utils";
+import { ClickAwayListener, Grow, Popper } from "@material-ui/core";
+import React, { useMemo, useState } from "react";
+
+import { FilterButton } from "./FilterButton";
+import { useStyles } from "./styles";
+
+export interface FilterProps<TFilterKeys extends string = string> {
+  currencySymbol?: string;
+  errorMessages?: FilterErrorMessages<TFilterKeys>;
+  menu: IFilter<TFilterKeys>;
+  onFilterAdd: (filter: Array<FilterElement<string>>) => void;
+  onFilterAttributeFocus?: (id?: string) => void;
+}
+
+const Filter = ({
+  currencySymbol,
+  menu,
+  onFilterAdd,
+  onFilterAttributeFocus,
+  errorMessages,
+}) => {
+  const classes = useStyles();
+
+  const anchor = React.useRef<HTMLDivElement>();
+  const [isFilterMenuOpened, setFilterMenuOpened] = useState(false);
+  const [filterErrors, setFilterErrors] = useState<InvalidFilters<string>>({});
+  const [data, dispatch, reset] = useFilter(menu);
+
+  const isFilterActive = menu.some(filterElement => filterElement.active);
+  const selectedFilterAmount = useMemo(
+    () =>
+      menu.reduce((acc, filterElement) => {
+        const dataFilterElement = data.find(
+          ({ name }) => name === filterElement.name,
+        );
+
+        if (!dataFilterElement) {
+          return acc;
+        }
+
+        return acc + (dataFilterElement.active ? 1 : 0);
```

Comment:
**suggestion**: it deserves dedicated function, hard to read what this memo is doing